web/wiki/src/__tests__/organisms/CardResource.test.tsx

```diff
@@ -98,4 +138,140 @@ describe('CardResource component', () => {
     const favIcon = screen.getByTitle('Afegeix a preferits')
     expect(favIcon).toBeInTheDocument()
   })
+
+  it('cannot vote when user is not logged in', async () => {
+    vi.mocked(useAuth).mockReturnValue({
+      user: null,
+    } as TAuthContext)
+
+    render(<CardResource {...mockCardResource} />)
+
+    const upvoteBtn = screen.getByTestId('increase')
+    fireEvent.click(upvoteBtn)
+    await waitFor(() => {
+      expect(handleAccessModal).toHaveBeenCalled()
+    })
+  })
+
+  it('can vote when the user is logged in, with getResources', async () => {
+    vi.mocked(useAuth).mockReturnValue({
+      user: {
+        name: 'Test author name',
+        avatarId: 'profileAvatar.jpg',
+      },
+    } as TAuthContext)
+
+    queryClient.setQueryData(['getResources'], [mockCardResource.id])
+    const queryData = queryClient.getQueryData(['getResources']) as TResource[]
+
+    const { rerender } = render(
+      <CardResource voteCount={queryData[0].voteCount} {...mockCardResource} />
+    )
+
+    const upvoteBtn = screen.getByTestId('increase')
+    const downvoteBtn = screen.getByTestId('decrease')
+    expect(upvoteBtn).toBeInTheDocument()
+    expect(downvoteBtn).toBeInTheDocument()
+    expect(screen.getByText('124')).toBeInTheDocument()
+
+    fireEvent.click(upvoteBtn)
+
+    await waitFor(() => {
+      expect(handleClick).toHaveBeenCalled()
+      expect(screen.getByText('125')).toBeInTheDocument()
+    })
+    await waitFor(() => {
+      const queryDataUpdated = queryClient.getQueryData([
+        'getResources',
+      ]) as TResource[]
+
+      rerender(
+        <CardResource
+          voteCount={queryDataUpdated[0].voteCount}
+          {...mockCardResource}
+        />
```

Comment:
You're almost there! But the mock is overwriting the voteCount, just invert the order